### PR TITLE
Fix #3758: Regression test for #3596

### LINKED ISSUE #3758: Automated tests for the incremental compiler/linker
See https://github.com/scala-js/scala-js/pull/3757#pullrequestreview-282154854

### PR DESCRIPTION
**Pull request by gzm0**

### CODE REVIEW

**Comment by gzm0** [comment]
@sjrd so the tip about the missing module initializer was good, now it always fails :-/

**Comment by gzm0** [comment]
Just FTR, this does not need to go into RC1 IMHO.

**Comment by gzm0** [comment]
Also note: #3878

File: sbt-plugin/src/sbt-test/incremental/fix-compile-error/Data.scala.fixed~ [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package org.scalajs.sbtplugin.test
```
Comment (sjrd) [inline sbt-plugin/src/sbt-test/incremental/fix-compile-error/Data.scala.fixed~]: This `...~` file shouldn't be committed ;)

**Comment by gzm0** [comment]
Updated.

**Pull request merged**